protocols/floodsub/src/protocol.rs

```diff
@@ -58,8 +88,20 @@ where
 
     fn upgrade_inbound(self, mut socket: TSocket, _: Self::Info) -> Self::Future {
         Box::pin(async move {
-            let packet = upgrade::read_length_prefixed(&mut socket, 2048).await?;
-            let rpc = rpc_proto::Rpc::decode(&packet[..]).map_err(DecodeError)?;
+            // From Thomas:
+            // What you probably want to do here is use a Framed to wrap
+            // the incoming socket and immediately read a message from it
+            // using its Stream::next implementation.
+            //
+            // Framed is an abstraction on top of a codec to
+            // read and write from a socket in entire messages instead of
+            // raw bytes!
+
+            Framed::new(socket, FloodsubProtocol::new());
+            let packet = upgrade::read_length_prefixed(&mut socket, MAX_MESSAGE_LEN_BYTES).await?;
+            // Replace with prost_codec::Codec
+            let rpc = self.codec.decode();
+            // prost_codec::Codec::<rpc_proto::Rpc>::decode(&packet[..]).map_err(DecodeError)?;
```

Comment:
```suggestion
            let mut framed = Framed::new(socket, FloodsubProtocol::new());
            let rpc = framed.next().await; // TODO Needs error handling
```

This should do the trick!